lib/omise/OmiseCapabilities.php

```diff
@@ -0,0 +1,180 @@
+    {
+        return parent::g_retrieve(get_class(), self::getUrl(), $publickey, $secretkey);
+    }
+
+    /**
+     * Retrieves array of payment backends. Optionally pass in as many filter functions as you want
+     * (muliple arguments, or a single array)
+     *
+     * @param [func1,fun2,...] OR func1, func2,...
+     *    
+     * @return array
+     */
+    public function getBackends() {
+        // check for filters
+        if ($filters = func_get_args()) $filter = self::_combineFilters(self::_fixArgs($filters));
+        $res = $this['payment_backends'];
+        array_walk(
+            $res,
+            function($v, $k) use (&$res) {
+                $id = array_keys($v)[0];
+                $res[$k][$id]['_id'] = $id;
+            }
+        );
+        $res = array_map(function($a) { return (object)reset($a);}, $res);
+        return !empty($filter) ? array_filter($res, $filter) : $res;
+    }
+
+    /**
+     * Makes a filter function to check supported currency for backend.
+     *
+     * @param  string $currency
+     *
+     * @return function
+     */
+    public function makeBackendFilterCurrency($currency) {
+        return function($backend) use ($currency) { return in_array(strtoupper($currency), $backend->currencies); };
+    }
+
+    /**
+     * Makes a filter function to check type of backend.
+     *
+     * @param  string $type
+     *
+     * @return function
+     */
+    public function makeBackendFilterType($type) {
+        return function($backend) use ($type) { return $backend->type==$type; };
+    }
+
+    /**
+     * Makes a filter function to check if backends can handle given amount.
+     *
+     * @param  int $amount
+     *
+     * @return function
+     */
+    public function makeBackendFilterChargeAmount($amount) {
+        $defMin = $this['limits']['charge_amount']['min'];
+        $defMax = $this['limits']['charge_amount']['max'];
+        return function($backend) use ($amount, $defMin, $defMax) {
+            // temporary hack for now to correct min value for instalments to 500000
+            if ($backend->type == 'installment') {
+                $min = 500000;
+            } else {
+                $min = empty($backend->amount['min']) ? $defMin : $backend->amount['min'];
+            }
+            $max = empty($backend->amount['max']) ? $defMax : $backend->amount['max'];
+            return $amount >= $min && $amount <= $max;
+        };
+    }
+
+    /**
+     * Combines boolean filters.
+     *
+     * @param  [functions] $filters
+     *
+     * @return function
+     */
+    private static function _combineFilters($filters) {
+        return function($a) use ($filters) {
+            foreach ($filters as $filter) if (!$filter($a)) return false;
+            return true;
+        };
+    }
+
+    /**
+     * Combines boolean filters.
+     *
+     * @param  [functions] $filters
+     *
+     * @return function
+     */
+    private static function _fixArgs($argArray) {
```

Comment:
@jonrandy can we have a better name for this one? `fixArgs` sounds a bit ambiguous..

Reply:
`_sanitiseArgs`?

Reply:
Have gone with `_argsToVariadic`